docs/tutorial/index.org

```diff
@@ -0,0 +1,1112 @@
+associated to a cluster by consulting the [[https://www.grid5000.fr/mediawiki/images/G5k_cheat_sheet.pdf][G5k Cheatsheet]]. If you are
+concerned, connect to the network node as root with:
+#+BEGIN_SRC sh
+# (venv) enos:~/enos-myxp$
+ssh -l root $(enos info --out json | jq -r '.rsc.network[0].address')
+#+END_SRC
+
+And execute the following script:
+#+BEGIN_SRC sh  :export code
+#!/usr/bin/env bash
+
+# The network interface
+IF=eth0
+# This is the list of the vip of $IF
+ips=$(ip addr show dev $IF|grep "inet .*/32" | awk '{print $2}')
+if [[ ! -z "$ips" ]]
+then
+  # vip detected
+  echo $ips
+  docker exec -ti openvswitch_vswitchd ovs-vsctl add-port br-ex $IF && ip addr flush $IF && dhclient -nw br-ex
+  for ip in $ips
+  do
+    ip addr add $ip dev br-ex
+  done
+else
+  echo "nothing to do"
+fi
+#+END_SRC
+#+END_note
+
+Then, display the information about your VM with the following command:
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+# (venv) enos:~/enos-myxp$
+openstack server show cli-vm
+#+END_SRC
+
+Note in particular the status of your VM. This status will go from
+~BUILD~: OpenStack is looking for the best place to boot the VM, to
+~ACTIVE~: your VM is running. The status could also be ~ERROR~ if you
+are experiencing hard times with your infrastructure.
+
+With the previous ~openstack server create~ command, the VM boots with
+a private IP. Private IPs are used for communication between VMs,
+meaning you cannot ping your VM from the lab machine. Network lovers
+will find a challenge here: try to ping the VM from the lab machine.
+For the others, you have to manually affect a floating IP to your
+machine if you want it pingable from the enos node.
+
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+# (venv) enos:~/enos-myxp$
+openstack server add floating ip\
+  cli-vm\
+  $(openstack floating ip create public -c floating_ip_address -f value)
+#+END_SRC
+
+You can ask for the status of your VM and its IPs with:
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+# (venv) enos:~/enos-myxp$
+openstack server show cli-vm -c status -c addresses # (venv) enos-node$
+#+END_SRC
+
+Wait one minute or two the time for the VM to boot, and when the state
+is ~ACTIVE~, you can ping it on its floating IP and SSH on it:
+
+#+BEGIN_SRC sh
+# (venv) enos:~/enos-myxp$
+ping <floating-ip>
+# (venv) enos:~/enos-myxp$
+ssh -l cirros <floating-ip>
+#+END_SRC
+
+#+BEGIN_note
+Waiting for the IP to appear and then ping it could be done with a
+bunch of bash commands, such as in listing [[lst:query-ip]].
+
+#+CAPTION: Query the ~cli-vm~, wait until the floating IP is affected,
+#+CAPTION: and finally ping it.
+#+NAME: lst:query-ip
+#+BEGIN_SRC sh :tangle ../../tests/functionnal/tests/tutorial/enos-node.sh
+FLOATING_IP="openstack server show cli-vm -c addresses -f value | sed  -Er 's/private=(10\.0\.0\.[0-9]+).*/\1/g'"
```

Comment:
We can use `openstack floating ip` command to get the floating ips.
`FLOATING_IP="openstack floating ip list -c 'Floating IP Address' -f value"`

Note: This work under the assumption that there's only one floating ip associated with a VM. Otherwise we can add some filter.

Reply:
I fixed it like this: 
```sh
FIXED_IP=$(openstack server show cli-vm -c addresses -f value | sed  -Er 's/private=(10\.0\.0\.[0-9]+).*/\1/g')
FLOATING_IP=$(openstack floating ip list --fixed-ip-address "$FIXED_IP" -f value -c "Floating IP Address" | head -n 1)
```